docs/src/modules/ROOT/pages/using-timefold-solver/running-the-solver.adoc

```diff
@@ -180,76 +209,78 @@ Any variable that changed between the "before move" solution and the "after move
+
+This mode is <<environmentModeReproducibility,reproducible>>.
+It is also intrusive because it calls the method `calculateScore()` more frequently than a non-assert mode.
+
+The `FULL_ASSERT` mode is horribly slow,
+because it does not rely on incremental score calculation.
+
+
 [#environmentModeNonIntrusiveFullAssert]
-=== `NON_INTRUSIVE_FULL_ASSERT`
+==== `NON_INTRUSIVE_FULL_ASSERT`
 
 The `NON_INTRUSIVE_FULL_ASSERT` turns on several assertions to fail-fast on a bug in a Move implementation,
 a constraint, the engine itself, ...
 
-This mode is <<environmentModeReproducible,reproducible>>.
+This mode is <<environmentModeReproducibility,reproducible>>.
 It is non-intrusive because it does not call the method `calculateScore()` more frequently than a non-assert mode.
 
 The `NON_INTRUSIVE_FULL_ASSERT` mode is horribly slow,
 because it does not rely on incremental score calculation.
 
 
 [#environmentModeStepAssert]
-=== `STEP_ASSERT`
+==== `STEP_ASSERT`
 
 The `STEP_ASSERT` mode turns on most assertions (such as assert that an undoMove's score is the same as before the Move)
 to fail-fast on a bug in a Move implementation, a constraint, the engine itself, ...
+This makes it slow.
 
-This mode is <<environmentModeReproducible,reproducible>>.
+This mode is <<environmentModeReproducibility,reproducible>>.
 It is also intrusive because it calls the method `calculateScore()` more frequently than a non-assert mode.
 
-The `STEP_ASSERT` mode is slow.
-
-It is recommended to write a test case that does a short run of your planning problem with the `STEP_ASSERT` mode on.
+We recommend that you write a test case that does a short run of your planning problem with the `STEP_ASSERT` mode on.
 
 
 [#environmentModePhaseAssert]
-=== `PHASE_ASSERT` (default)
+==== `PHASE_ASSERT` (default)
 
 The `PHASE_ASSERT` is the default mode because it is recommended during development.
-In this mode, two runs in the same Timefold Solver version will execute the same code in the same order.
-**Those two runs will have the same result at every step**, except if the note below applies.
-This enables you to reproduce bugs consistently.
-It also allows you to benchmark certain refactorings (such as a score constraint performance optimization) fairly across runs.
+This mode is <<environmentModeReproducibility,reproducible>>
+and is negligibly slower than the `NO_ASSERT` mode.
+However, it gives you the benefit of quickly checking for score corruptions.
+If you can guarantee that your code is and will remain bug-free,
+you can switch to the `NO_ASSERT` mode for a marginal performance gain.
 
-[NOTE]
-====
-Despite the reproducible mode, your application might still not be fully reproducible because of:
+In practice, this mode uses the default, fixed <<randomNumberGenerator,random seed>> if no seed is specified,
```

Comment:
Since Seeds aren't really mentioned clearly before, I wouldn't introduce the concept here.